lib/Fmt_ast.ml

```diff
@@ -2662,12 +2662,7 @@ and fmt_class_signature c ~ctx ~parens ?ext self_ fields =
     | Pctf_attribute atr -> update_config c [atr]
     | _ -> c
   in
-  let cmts_after_self = Cmts.fmt_after c self_.ptyp_loc in
-  let self_ =
-    match self_ with
-    | {ptyp_desc= Ptyp_any; ptyp_attributes= []; _} -> None
-    | s -> Some s
-  in
+  let cmts_after_self = Cmts.fmt_after c self_.pcss_loc in
```

Comment:
The issue you might see here is that ocamlformat will attach comments to the closest location, so some comments might be previously attached to `self_.ptyp_loc` and moved to `self_.pcss_loc`.

And you don't print the comments located *before* the location `self_.pcss_loc`. It was not an issue with `self_.ptyp_loc` because when calling `fmt_core_type` we would print the remaining comments attached to this type anyway, not the case with the new loc field.

I dont't think adding a new location makes much sense here, as I've said in the parsetree comment, when the type is None the location should not exist anyway, so imho the location of the existing type is enough.